### ISSUE TITLE
Cmake module does not sanitize variable names when generating meson.build

### ISSUE THREAD

**Issue by albert-chang0** [mesonbuild/meson#6554]
**Describe the bug**
If a subproject that uses CMake names a library with something outside of alphanumeric and underscore characters, the meson module does not sanitize the name, resulting in a failed build.

For example if the `CMakeLists.txt` file contains:
```
...
add_library(grpc++
  src/cpp/client/insecure_credentials.cc
...
  src/cpp/codegen/codegen_init.cc
)
set_target_properties(grpc++ PROPERTIES
  VERSION ${gRPC_CPP_VERSION}
  SOVERSION ${gRPC_CPP_SOVERSION}
)
...
```

**To Reproduce**
Encountered this with gRPC:

```
$ cd subprojects
$ git submodule add https://github.com/grpc/grpc.git
$ git submodule sync --recursive
$ git submodule update --init --recursive -- grpc
$ meson out
$ ninja -C out
...

subprojects/grpc/meson.build:0:0: ERROR: Invalid variable name: cm_grpc++_dir

A full log can be found at /mnt/data/extra/achang/src/chillfs/out/meson-logs/meson-log.txt
FAILED: build.ninja
```

Here's a snippet of `meson.build` to include it:
```
cmake = import('cmake')
grpc_proj = cmake_native.subproject('grpc')
grpc_dep = grpc_proj.dependency('grpc_dep')
deps += grpc_dep
executable(
  'myprogram',
  dependencies : deps,
  install : true
)
```

**Expected behavior**
Can build the library and use it as a subproject.

**system parameters**
* Is this a [cross build](https://mesonbuild.com/Cross-compilation.html) or just a plain native build (for the same computer)? **plain native build**
* what operating system (e.g. MacOS Catalina, Windows 10, CentOS 8.0, Ubuntu 18.04, etc.): **Archlinux, but likely an issue everywhere**
* what Python version are you using? **3.8.1**
* what `meson --version`? **0.53.0**
* what `ninja --version` if it's a Ninja build? **1.9.0**

**Comment by Berticus** [comment]
Might be fixed with #6213

**Comment by mensinda** [comment]
Should be fixed with #6555
